pkg/notifier/beans/beans.go

```diff
@@ -0,0 +1,394 @@
+package beans
+
+import (
+	util "github.com/devtron-labs/devtron/util/event"
+)
+
+type NotificationConfigRequest struct {
+	Id int `json:"id"`
+
+	TeamId    []*int `json:"teamId"`
+	AppId     []*int `json:"appId"`
+	EnvId     []*int `json:"envId"`
+	ClusterId []*int `json:"clusterId"`
+
+	PipelineId   *int              `json:"pipelineId"`
+	PipelineType util.PipelineType `json:"pipelineType" validate:"required"`
+	EventTypeIds []int             `json:"eventTypeIds" validate:"required"`
+	Providers    []*Provider       `json:"providers"`
+}
+
+func (notificationSettingsRequest *NotificationConfigRequest) GenerateSettingCombinationsV1() []*LocalRequest {
```

Comment:
this func seems like part of a util file

Reply:
since this function is defined on the struct, we should keep it here itself.